src/routes/transactions/mappers/transactions-history.mapper.ts

```diff
@@ -162,16 +171,18 @@ export class TransactionsHistoryMapper {
   }
 
   private mapTransfer(transfers: Transfer[], chainId: string, safe: Safe) {
-    return transfers.map(
-      async (transfer) =>
-        new TransactionItem(
-          await this.incomingTransferMapper.mapTransfer(
-            chainId,
-            transfer,
-            safe,
+    return transfers
+      .slice(0, this.maxNestedTransfers)
```

Comment:
We should be careful regarding how this variable is tied to the `slice` function as negative numbers are allowed.

For example `slice(0, -2)` would give the last two transfers. While this is part of the configuration we should bear in mind the values that we allow.

Reply:
Yes, good point 👍